hub_streamlit/src/infrastructure/data/edge_data.py

```diff
@@ -0,0 +1,96 @@
+import json
+from datetime import datetime
+from typing import Dict, List, Optional
+
+from google.api_core.exceptions import NotFound
+from google.cloud.storage import Blob, Bucket
+from pydantic import BaseModel, Field
+from PIL import Image
+
+
+def read_edge_ip(bucket: Bucket, edge_name: str) -> Optional[str]:
+    blob = bucket.blob(f"{edge_name}/edge_ip.txt")
+    try:
+        edge_ip = blob.download_as_text()
+    except NotFound as e:
+        print(f"Edge IP not found for {edge_name}. Error: {e}")
+        edge_ip = None
+    return edge_ip
+
+
+def read_metadata(
+    bucket: Bucket, edge_name: str, use_case: str, item_id: str
+) -> Optional[dict]:
+    blob = bucket.blob(f"{edge_name}/{use_case}/{item_id}/metadata.json")
+    if blob.exists():
+        metadata = json.loads(blob.download_as_text())
+    else:
+        metadata = None
```

Comment:
on peut garder la même structure que `read_edge_ip` : 
```suggestion
    try:
        metadata = json.loads(blob.download_as_text())
    except NotFound as e:
        metadata = None
```